### PR TITLE
286 empty plugin raises unhelpful error

### LINKED ISSUE #286: When a plugin file is empty an unhelpful error is raised
# Requested Update

### Description
When a plugin file contains nothing, it results in yaml that just contains `None`. This is not well handled and results in the following error:

```
In [1]: from geoips import interfaces
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 from geoips import interfaces

File /local/home/jsolbrig/geoips/github/geoips/geoips/__init__.py:31
     29 from geoips import errors
     30 from geoips import filenames
---> 31 from geoips import interfaces
     32 from geoips import utils
     33 from geoips import xarray_utils

File /local/home/jsolbrig/geoips/github/geoips/geoips/interfaces/__init__.py:30
     27 from geoips.interfaces.module_based.sector_spec_generators import sector_spec_generators
     28 from geoips.interfaces.module_based.title_formatters import title_formatters
---> 30 from geoips.interfaces.yaml_based.feature_annotators import feature_annotators
     31 from geoips.interfaces.yaml_based.gridline_annotators import gridline_annotators
     32 from geoips.interfaces.yaml_based.product_defaults import product_defaults

File /local/home/jsolbrig/geoips/github/geoips/geoips/interfaces/yaml_based/feature_annotators.py:24
     19     """Interface for feature annotator plugins."""
     21     name = "feature_annotators"
---> 24 feature_annotators = FeatureAnnotatorsInterface()

File /local/home/jsolbrig/geoips/github/geoips/geoips/interfaces/base.py:438, in BaseYamlInterface.__init__(self)
    435 def __init__(self):
    436     """YAML plugin interface init method."""
    437     self._unvalidated_plugins = self._create_unvalidated_plugins_cache(
--> 438         load_all_yaml_plugins()
    439     )

File /local/home/jsolbrig/geoips/github/geoips/geoips/geoips_utils.py:93, in load_all_yaml_plugins()
     89 yaml_plugin = yaml.safe_load(open(yaml_file, "r"))
     91 # Set some additional information on the YAML plugin
     92 # The name of the package the plugin comes from
---> 93 yaml_plugin["package"] = pkg.value
     94 # The relative path to the plugin within the package
     95 yaml_plugin["relpath"] = str(yaml_file.relative_to(pkg_plugin_path))

TypeError: 'NoneType' object does not support item assignment
```

### Background and Motivation
*  [ OPTIONAL: Provide additional detailed information and background related to this issue ]

### Alternative Solutions
*  [ OPTIONAL: If appropriate, list alternative fixes that could be considered]
  
### Environment
*  [ IF APPROPRIATE: Corresponding repository versions ]

### Code to demonstrate issue
*  [ IF APPROPRIATE: Can include steps here to identify/demonstrate bug fix or desired feature ]

### Checklist for Completion
* [ ]

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [x] Existing tests pass (full_test.sh passes)
* [x] Appropriate unit tests added and pass for new/modified functionality   
  * [x] No unit tests required
* [ ] Appropriate integration tests added and pass for new/modified functionality
  * [x] No integration tests required.
* [ ] Appropriate documentation added for new/modified functionality
  * [x] No documentation required
* [x] Appropriate release notes added for new/modified functionality

# Related Issues
fixes #286 

# Testing Instructions
To test this, create an empty YAML plugin, then open 'ipython' and run the command:
                `from geoips import interfaces`
This will result in the output shown below in the 'output' section.

# Summary
Currently, when a YAML plugin is empty in GeoIPS, the error raised is non-descriptive
and not helpful. To fix this, we've added a PluginError which points the user to the
empty YAML file, and tells them to fill it with the appropriate information.

   -  modified: geoips/geoips/geoips_utis.py

# Output
Output from running the command `from geoips import interfaces` in 'ipython'.

Screenshot 2023-09-07 at 10 33 07 AM

### CODE REVIEW

**Review by mindyls — APPROVED** [review]
I merged 4 branches into a single local "test_changes" branch and successfully ran "full_test.sh" on all:

https://github.com/NRLMMD-GEOIPS/geoips/pull/319
https://github.com/NRLMMD-GEOIPS/geoips/pull/320
https://github.com/NRLMMD-GEOIPS/geoips/pull/316
https://github.com/NRLMMD-GEOIPS/geoips/pull/314

Note the print statement PR required a single update (log->LOG in AHI reader), otherwise these all passed without modification.

I am good merging all of these PRs.

Package: geoips_full
Total run time: 2416 seconds
Number data types run: 47
Number data types failed: 0

File: geoips/geoips_utils.py [hunk 1]
```diff
@@ -90,6 +90,11 @@ def load_all_yaml_plugins():
 
             # Set some additional information on the YAML plugin
             # The name of the package the plugin comes from
+            if yaml_plugin is None:
```
Comment (mindyls) [inline geoips/geoips_utils.py:93]: Should this be if yaml_plugin rather than is None?

File: docs/source/releases/v1_11_3a0.rst [hunk 2]
```diff
@@ -60,6 +61,17 @@ has been resolved.
...
+Add PluginError for empty YAML plugins
+--------------------------------------
+
+Currently, when a YAML plugin is empty in GeoIPS, the error raised is non-descriptive
+and not helpful. To fix this, we've added a PluginError which points the user to the
+empty YAML file, and tells them to fill it with the appropriate information.
+
+::
+
+    modified: geoips/geoips/geoips_utis.py
```
Comment (mindyls) [inline docs/source/releases/v1_11_3a0.rst:73]: Do we want to add a test for this?

Comment (jsolbrig) [reply docs/source/releases/v1_11_3a0.rst:116]: What kind of test are you thinking here?

Comment (mindyls) [reply docs/source/releases/v1_11_3a0.rst:116]: An empty YAML file, but I do not really care.  Just want to make sure we are on the same page :-)

**Review by jsolbrig — APPROVED** [review]

File: geoips/geoips_utils.py (hunk 1 above)
Comment (jsolbrig) [reply geoips/geoips_utils.py]: I updated this. It's correct to test that it is false so it is now `if not yaml_plugin:`.

File: docs/source/releases/v1_11_3a0.rst (hunk 2 above)
Comment (jsolbrig) [reply docs/source/releases/v1_11_3a0.rst:116]: I think it's good for now.

**Pull request merged**